ui-spacetimechart/src/components/workSchedules/Region.tsx

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useDraw } from '../../hooks/useCanvas';
+import { type DrawingFunction } from '../../lib/types';
+import { type SpaceTimeRegion } from '../../types';
+
+const Region = ({
```

Comment:
I wonder if we can find a better name, "Region" seems a bit generic to me. What this component truly does is paint a rectangle with an image pattern. What about "PatternRect"?

Reply:
done https://github.com/OpenRailAssociation/osrd-ui/pull/523#discussion_r1787379091